vite-plugin-alphatab-copy: path computing issue on github pages / maybe Ubuntu in general

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I'm trying to deploy an app containing alphaTab on github pages, which is built using vite / alphaTab vite plugin.

On github pages the build fails while the `vite-plugin-alphatab-copy` plugin is trying to copy some assets into the public folder.
https://github.com/bali182/alphatab-vite-build-bug-repro/actions/runs/9906924688/job/27369512708

Messages are like this: 

```
[vite-plugin-alphatab-copy] ENOTDIR: not a directory, copyfile '/home/runner/work/stick/stick/node_modules/@coderline/alphatab/dist/soundfont/LICENSE/LICENSE' -> '/home/runner/work/stick/stick/public/soundfont/LICENSE'
Error: ENOTDIR: not a directory, copyfile '/home/runner/work/stick/stick/node_modules/@coderline/alphatab/dist/soundfont/LICENSE/LICENSE' -> '/home/runner/work/stick/stick/public/soundfont/LICENSE'
Error: Process completed with exit code 1.
```

The file names are doubly present in the path its trying to copy from:
```'/home/runner/work/stick/stick/node_modules/@coderline/alphatab/dist/soundfont/LICENSE/LICENSE'```

To see what's happening, I made a simplified version of `copyAssetsPlugin.ts` and when running it as an individual plugin and logging the  `files` in `copyFiles`, it seems like on gh pages (maybe on Linux on general, IDK) the `readdir` functions `file` already has the full file path. So this:

```ts
await Promise.all(
  files
    .filter((f) => f.isFile())
    .map(async (file) => {
      const sourceFilename = path.join(file.path, file.name)
      console.log({
        'file.path': file.path,
        'file.name': file.name,
        sourceFilename: sourceFilename,
      })
      await fs.promises.copyFile(
        sourceFilename,
        path.join(outputPath!, subdir, file.name),
      )
    }),
)
```

Results in this:

```js
{
  'file.path': '/home/runner/work/stick/stick/node_modules/@coderline/alphatab/dist/font/Bravura-OFL-FAQ.txt',
  'file.name': 'Bravura-OFL-FAQ.txt',
  sourceFilename: '/home/runner/work/stick/stick/node_modules/@coderline/alphatab/dist/font/Bravura-OFL-FAQ.txt/Bravura-OFL-FAQ.txt'
}
{
  'file.path': '/home/runner/work/stick/stick/node_modules/@coderline/alphatab/dist/font/Bravura-OFL.txt',
  'file.name': 'Bravura-OFL.txt',
  sourceFilename: '/home/runner/work/stick/stick/node_modules/@coderline/alphatab/dist/font/Bravura-OFL.txt/Bravura-OFL.txt'
}
```

And `file.path` has the full path of the file. So appending the file name again will result in double file names.


### Expected Behavior

The build to succeed

### Steps To Reproduce

Here is a simple repro to reproduce this with the latest vite and alphatab versions:
https://github.com/bali182/alphatab-vite-build-bug-repro

And here are the logs of a failed run
https://github.com/bali182/alphatab-vite-build-bug-repro/actions/runs/9906924688/job/27369512708

### Link to jsFiddle, CodePen, Project

https://github.com/bali182/alphatab-vite-build-bug-repro

### Found in Version

1.4 (alpha)

### Platform

Web

### Environment

```markdown
- **OS**: ubuntu latest
- **Browser**: not relevant
- **.net Version**: not relevant
```


### Anything else?

_No response_

--- a/package.json
+++ b/package.json
@@ -109,6 +109,7 @@
     ],
     "dependencies": {
         "@types/express": "^4.17.21",
+        "@types/node": "^20.14.10",
         "@types/opener": "^1.4.3"
     }
 }

--- a/src/vite/copyAssetsPlugin.ts
+++ b/src/vite/copyAssetsPlugin.ts
@@ -62,7 +62,9 @@ export function copyAssetsPlugin(options: AlphaTabVitePluginOptions): Plugin {
                     files
                         .filter(f => f.isFile())
                         .map(async file => {
-                            const sourceFilename = path.join(file.path, file.name);
+                            // node v20.12.0 has parentPath pointing to the path (not the file)
+                            // see https://github.com/nodejs/node/pull/50976
+                            const sourceFilename = path.join(file.parentPath ?? file.path, file.name);
                             await fs.promises.copyFile(sourceFilename, path.join(outputPath!, subdir, file.name));
                         })
                 );

--- a/src/webpack/AlphaTabWebPackPlugin.ts
+++ b/src/webpack/AlphaTabWebPackPlugin.ts
@@ -144,7 +144,8 @@ export class AlphaTabWebPackPlugin {
             alphaTab: {} as any
         } satisfies webPackWithAlphaTab;
 
-        if ('alphaTab' in compiler.webpack.util.serialization.register) { // prevent multi registration
+        if ('alphaTab' in compiler.webpack.util.serialization.register) {
+            // prevent multi registration
             webPackWithAlphaTab.alphaTab = compiler.webpack.util.serialization.register.alphaTab;
         } else {
             (compiler.webpack.util.serialization.register as any).alphaTab = webPackWithAlphaTab.alphaTab;
@@ -262,7 +263,9 @@ export class AlphaTabWebPackPlugin {
                         files
                             .filter(f => f.isFile())
                             .map(async file => {
-                                const sourceFilename = path.join(file.path, file.name);
+                                // node v20.12.0 has parentPath pointing to the path (not the file)
+                                // see https://github.com/nodejs/node/pull/50976
+                                const sourceFilename = path.join(file.parentPath ?? file.path, file.name);
                                 await fs.promises.copyFile(sourceFilename, path.join(outputPath!, subdir, file.name));
                                 const assetFileName = subdir + '/' + file.name;
                                 const existingAsset = compilation.getAsset(assetFileName);

--- a/test/bundler/Vite.test.ts
+++ b/test/bundler/Vite.test.ts
@@ -49,7 +49,7 @@ describe('Vite', () => {
 
         for (const file of dir) {
             if (file.isFile()) {
-                const text = await fs.promises.readFile(path.join(file.path, file.name), 'utf8');
+                const text = await fs.promises.readFile(path.join(file.parentPath ?? file.path, file.name), 'utf8');
 
                 if (file.name.startsWith('index-')) {
                     // ensure new worker has worker import

--- a/test/bundler/WebPack.test.ts
+++ b/test/bundler/WebPack.test.ts
@@ -83,7 +83,7 @@ describe('WebPack', () => {
 
         for (const file of dir) {
             if (file.isFile()) {
-                const text = await fs.promises.readFile(path.join(file.path, file.name), 'utf8');
+                const text = await fs.promises.readFile(path.join(file.parentPath ?? file.path, file.name), 'utf8');
 
                 if (file.name.startsWith('app-')) {
                     // ensure new worker has worker import

